### PR TITLE
Feature/jpaley geom align

### PR DESCRIPTION
**Pull request by jonpaley**
This is mostly an overhaul of the Geometry package.  Some changes to the EventDisplay.  Other packages were updated to work with the changes in the Geometry class.

I have tested that simulation, digitization, and clustering work fine.  I have not tested Robert's  TrackReco package.  I'd like to pull the trigger on this ASAP, as it could impact on-going work by others, and they should adapt to these changes before it becomes a lot of work to do so...

### CODE REVIEW

File: Geometry/Geometry.h [hunk 1]
```diff
@@ -25,173 +27,210 @@ class TGeoManager;
...
+    class Strip {
+    public:
+      Strip();
+      Strip(std::string name, TVector3 pos, double dw);
+      ~Strip() {};
+
+      std::string Name() const { return fName;}
+      TVector3 Pos() const { return fPos;}
+      double Dw() const { return fDw;}
+      void LocalToMaster(double x1[3], double x2[3]) const { fGeoMatrix->LocalToMaster(x1,x2); }
```
Comment (lackey32) [inline Geometry/Geometry.h:47]: On NOvA we made an effort a couple years ago to move away from using 'Master', I'd suggest 'Global' here.

File: SSDReco/MakeSSDClusters_module.cc [hunk 2]
```diff
@@ -57,16 +57,14 @@ class emph::MakeSSDClusters : public art::EDProducer {
...
+  int NPlanes;
   static const int NStations = 8;
-  static const int MaxSensPerSta = 6;
-  int ncluster[NPlanes];
-
-  std::vector<float> clustdist[NPlanes];
+  static const int MaxPlnsPerSta = 3;
+  static const int MaxSensPerPln = 2;
+  std::vector<int> ncluster;
+  //  std::vector<float> clustdist;
```
Comment (lackey32) [inline SSDReco/MakeSSDClusters_module.cc:66]: This was something Robert added, do we know that it doesn't have a use-case anymore? If so, the stuff that has been commented out related to it should be removed.

File: TrackReco/MakeHits_module.cc [hunk 3]
```diff
@@ -167,22 +166,25 @@ namespace emph {
   
   void MakeHits::beginRun(art::Run& run)
   {
-    art::ServiceHandle<emph::cmap::ChannelMapService> fChannelMap;
     art::ServiceHandle<emph::geo::GeometryService> geo;
     auto emgeo = geo->Geo();
+    nPlanes = emgeo->NSSDPlanes();
+    /*
```
Comment (lackey32) [inline TrackReco/MakeHits_module.cc:172]: Is there a reason to keep this code around?

File: ChannelMap/ChannelMap.cxx [hunk 4]
```diff
@@ -48,6 +48,17 @@ namespace emph {
 
     //----------------------------------------------------------------------
 
+    void ChannelMap::PrintE2D()
+    {
+      std::map<emph::cmap::EChannel,emph::cmap::DChannel>::iterator itr=fEChanMap.begin();
+      
+      for (; itr != fEChanMap.end(); ++itr) {
+	std::cout << itr->first << "-->" << itr->second << std::endl;
```
Comment (gavinsdavies) [inline ChannelMap/ChannelMap.cxx:56]: we should probably replace cout's with mf::LogInfo etc, so we can control the level of printouts in fcls. not worth holding up this PR though atm

**Review by gavinsdavies — APPROVED** [review]

**Review by lackey32 — CHANGES_REQUESTED** [review]
Just a handful of comments, mostly about removing commented out lines.

Have you tested this on data?

File: ChannelMap/ChannelMap.cxx (hunk 4 above)
Comment (jonpaley) [reply ChannelMap/ChannelMap.cxx:56]: Ok, I'll put that on my to-do list.  We should probably do this throughout our code...

File: ChannelMap/ChannelMap.cxx [hunk 5]
```diff
@@ -94,6 +105,8 @@ namespace emph {
 
       std::cout<<"Loaded channel map from " << fMapFileName << std::endl;
 
+      //      this->PrintE2D();
```
[the remark this answers is not in the document]
Comment (jonpaley) [reply ChannelMap/ChannelMap.cxx:108]: Removed

File: ChannelMap/ChannelMap.h [hunk 6]
```diff
@@ -120,8 +126,10 @@ namespace emph {
 	  if (fId == emph::geo::SSD) {
 	    if (fStation > dchan.fStation) return true;
 	    if (fStation == dchan.fStation) {
-	      if ((fChannel>=0) && (fChannel > dchan.fChannel)) return true;
-	      if ((fHiLo>=0) && (fHiLo > dchan.fHiLo)) return true;
+	      if (fPlane > dchan.fPlane) return true;
+	      if (fPlane == dchan.fPlane) 
+		if (fChannel > dchan.fChannel) return true;
+	      //	      if ((fHiLo>=0) && (fHiLo > dchan.fHiLo)) return true;
```
[the remark this answers is not in the document]
Comment (jonpaley) [reply ChannelMap/ChannelMap.h:132]: Done

File: ChannelMap/ChannelMap.h [hunk 7]
```diff
@@ -166,6 +178,9 @@ namespace emph {
       std::map<emph::cmap::EChannel,emph::cmap::DChannel> EChanMap() { return fEChanMap; }
       std::map<emph::cmap::DChannel,emph::cmap::EChannel> DChanMap() { return fDChanMap; }
 
+      void PrintE2D(); // print out map, echan -> dchan
+      //      void PrintD2E(); // print out map, dchan -> echan;
```
[the remark this answers is not in the document]
Comment (jonpaley) [reply ChannelMap/ChannelMap.h:182]: Forgot about that.  I've implemented it now.

File: EventDisplay/EventDisplay3D_module.cc [hunk 8]
```diff
@@ -491,7 +504,9 @@ void emph::EventDisplay3D::beginRun( const art::Run& )
 void emph::EventDisplay3D::drawSSDClust(Int_t mColor, Int_t mSize, 
 				      const rb::SSDCluster& cl)
 {
-
+  std::cout << "SSD cluster at (" << cl.Station() << ","
```
[the remark this answers is not in the document]
Comment (jonpaley) [reply EventDisplay/EventDisplay3D_module.cc:507]: Pretty noisy, I've removed them.

File: Geometry/Geometry.h (hunk 1 above)
Comment (jonpaley) [reply Geometry/Geometry.h:47]: Good point.  But Global is misleading, so I've changed it to "LocalToMother".  We still need to call Root's method (LocalToMaster) though.

File: SSDReco/MakeSSDClusters_module.cc (hunk 2 above)
Comment (jonpaley) [reply SSDReco/MakeSSDClusters_module.cc:66]: Yes, I've confirmed with Robert.  Removed.

File: TrackReco/MakeHits_module.cc (hunk 3 above)
Comment (jonpaley) [reply TrackReco/MakeHits_module.cc:172]: Nope, I just forgot about it.  I've removed it.

**Review by lackey32 — APPROVED** [review]
Thanks for the changes. We definitely should make an effort at some point to go through all the code and make sure things are well commented. Looks good to me to merge, assuming you've testing things with data.

**Pull request merged**